File: 941_Valid_Mountain_Array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        if len(arr) < 3:
            return False 
        increasing = True 
        if(arr[0] > arr[1]):
            return False 
        
        for i in range(1, len(arr)):
            if(arr[i] > arr[i-1]) and increasing:
                continue 
            elif(arr[i] == arr[i-1]):
                return False 
            elif(arr[i] < arr[i-1]) and increasing:
                increasing = False 
            elif(arr[i] < arr[i-1] and increasing == False):
                continue 
            elif(arr[i] > arr[i-1] and increasing == False):
                return False
        if(increasing):
            return False 
        return True 
                
                
        

File: 941_Valid_Mountain_Array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        if(arr.size() < 3) return false;
        bool increasing = true; 
        if(arr[0] > arr[1]) return false;
        for(int i = 1; i < arr.size(); i++){
            if((arr[i] > arr[i-1]) && increasing){
                continue; 
            }
            else if(arr[i] == arr[i-1]){
                return false;
            }
            else if(arr[i] < arr[i-1] && increasing){
                increasing = false; 
            }
            else if(arr[i] < arr[i-1] && increasing == false){
                continue;
            }
            else if(arr[i] > arr[i-1]  && increasing == false)
            {
                return false; 
            }
        }
        if(increasing) return false;
        return true; 
    }
};